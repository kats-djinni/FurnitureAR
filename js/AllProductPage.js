"use strict";

import React, { Component } from "react";

import {
  StyleSheet,
  View,
  Image,
  TouchableHighlight,
  FlatList
} from "react-native";
import { connect } from "react-redux";
import { Text } from "react-native-elements";
import { getAllProducts, pickProduct } from "./store/products"
import { getAllFavorites, storeFavorite, removeFavorite } from "./store/favorites"
import FavoriteButton from "./FavoriteButton";

export class AllProductPage extends Component {
  constructor() {
    super();
    this.state = {
      favorites: []
    };
  }
  async componentDidMount() {
    this.props.getProducts();
    this.props.getFavorites()
    .then(faves => {
      this.setState({favorites: faves})
    })
  }

  handlePress = event => {
    this.props.addPickedItem(event)
    this.props.visibilityChange()
  }

  filterFave = async item => {
    try {
      const faveArr = this.props.favorites
      if (faveArr !== null) {
        const duplicate = faveArr.filter(
          products => products.displayName === item.displayName
        )
        if (duplicate.length) {
          return true;
        } else {return false}
      } else {
        return false
      }
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    return (
      <View style={styles.listContainer}>
        <Text h4 style={styles.AllProductPage}>
          Choose Products
        </Text>
        <FlatList
          data={this.props.products}
          renderItem={({ item, index }) => (
            <View>
              <Text>Name: {item.displayName}</Text>
              <TouchableHighlight
                onPress={() => this.handlePress(item)}
                style={{ width: 200, height: 200 }}
              >
                <Image
                  style={{ width: 200, height: 200 }}
                  source={{ uri: item.thumbnail }}
                />
              </TouchableHighlight>
              <View style={styles.imageContainer}>
                <FavoriteButton
                  faveItem={item}
                  itemIndex={index}
                  active={this.filterFave(item)}
                />
              </View>
            </View>
          )}
          keyExtractor={(item, index) => index}
        />
      </View>
    );
  }
}

var styles = StyleSheet.create({
  AllProductPage: {
    fontFamily: "Arial",
    fontSize: 30,
    color: "#444B50",
    textAlignVertical: "center",
    textAlign: "center",
    margin: 5
  },
  listContainer: {
    flex: 1,
    flexDirection: "column",
    flexWrap: "wrap",
    fontFamily: "Arial",
    fontSize: 50,
    color: "#000000",
    textAlignVertical: "center",
    textAlign: "center",
    margin: 20
  },
  SingleItem: {
    flex: 1
  },
  imageContainer: {
    flex: 1
  }
});

const mapStateToProps = state => ({
  products: state.products.products,
  favorites: state.favorites.favorites
})

const mapDispatchToProps = dispatch => ({
  getProducts: () => dispatch(getAllProducts()),
  addPickedItem: item => dispatch(pickProduct(item)),
  getFavorites: () => dispatch(getAllFavorites()),
  addFavorite: (item) => dispatch(storeFavorite(item)),
  removeFavorite: (index) => dispatch(removeFavorite(index))
})

export default connect(mapStateToProps, mapDispatchToProps)(AllProductPage)
