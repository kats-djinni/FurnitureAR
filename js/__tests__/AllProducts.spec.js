import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { AllProductPage } from '../AllProductPage'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
enzyme.configure({ adapter: new Adapter() })
import { FlatList } from 'react-native'


describe('<AllProductPage /> component', () => {

    const products = [{
                "name": "assets/3txPAhYeu-x",
                "displayName": "Chair",
                "authorName": "Ryan D",
                "objurl": "https://poly.googleapis.com/downloads/fp/1551711596400293/3txPAhYeu-x/3DjXPukLzqV/model.obj",
                "mtlurl": "https://poly.googleapis.com/downloads/fp/1551711596400293/3txPAhYeu-x/3DjXPukLzqV/materials.mtl",
                "formatType": "OBJ",
                "thumbnail": "https://lh3.googleusercontent.com/dlwexhUXTVDfcmJayQRruP6IqK35UG1hw5C8Wmv-5E7d-cAfVQf5ThJ1cmiayHc"
            }
        ]

    let wrapper
    // const mockProductsfn = jest.fn()
 
    it('renders an FlatList', () => {
        const wrapper = shallow(<AllProductPage products={products} />)
        expect(wrapper.find(FlatList).length).toEqual(1)
      })

    it('renders correctly', () => {
        const wrapper = shallow(<AllProductPage products={products} />)
        expect(wrapper).toMatchSnapshot()
        })
})
