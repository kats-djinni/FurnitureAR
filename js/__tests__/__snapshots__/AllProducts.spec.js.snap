// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllProductPage /> component renders correctly 1`] = `ShallowWrapper {}`;
